### PR TITLE
Add Terraform provider documentation

### PR DESCRIPTION
**Pull request by kolchfa-aws**
Add Terraform provider documentation and fix redirect

### Checklist
- [x] By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and subject to the [Developers Certificate of Origin](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by kolchfa-aws** [comment]
@peterzhuamazon @prudhvigodithi This PR adds Terraform documentation to our site. Could one of you review when you get a chance please?

**Review by prudhvigodithi — APPROVED** [review]

**Comment by hdhalter** [comment]
Good catch on that redirect, Fanit!

File: _tools/terraform.md [hunk 1]
```diff
@@ -0,0 +1,10 @@
+---
+layout: default
+title: Terraform
+nav_order: 180
+has_children: false
+---
+
+# Terraform provider
+
+You can use the Terraform OpenSearch provider to provision OpenSearch resources and interact with the OpenSearch API. The Terraform provider supports AWS OpenSearch Service domains and OpenSearch clusters deployed on Kubernetes and other infrastructure. For more information, see the [Terraform provider documentation](https://registry.terraform.io/providers/opensearch-project/opensearch/latest/docs).
```
Comment (natebower) [inline _tools/terraform.md:10]:
```suggestion
You can use the Terraform OpenSearch provider to provision OpenSearch resources and interact with the OpenSearch API. The Terraform provider supports Amazon OpenSearch Service domains and OpenSearch clusters deployed on Kubernetes and other infrastructure. For more information, see the [Terraform provider documentation](https://registry.terraform.io/providers/opensearch-project/opensearch/latest/docs).
```

Comment (natebower) [inline _tools/terraform.md:10]: "other infrastructures" (plural)?

**Review by cwillum — APPROVED** [review]
LGTM

**Review by natebower — APPROVED** [review]
@kolchfa-aws Approved with the noted change/comment.

File: _tools/terraform.md (hunk 1 above)
Comment (kolchfa-aws) [reply _tools/terraform.md]: I think it makes sense to use "infrastructure" as a plural noun here.

**Pull request merged**